server/scsynth/SC_ComPort.cpp

```diff
@@ -258,13 +258,15 @@ class SC_UdpInPort {
         udpSocket.open(protocol);
 
         udpSocket.bind(ip::udp::endpoint(boost::asio::ip::address::from_string(bindTo), inPortNum));
+        if (inPortNum == 0)
+            mPortNum = udpSocket.local_endpoint().port();
 
         boost::asio::socket_base::send_buffer_size option(65536);
         udpSocket.set_option(option);
 
 #ifdef USE_RENDEZVOUS
         if (world->mRendezvous) {
-            SC_Thread thread(boost::bind(PublishPortToRendezvous, kSCRendezvous_UDP, sc_htons(mPortNum)));
+            SC_Thread thread(boost::bind(PublishPortToRendezvous, kSCRendezvous_UDP, mPortNum));
```

Comment:
Are we sure the byte swap is not necessary here? I "blamed" back and this seems to be like this since 2002 with its introduction via https://github.com/supercollider/supercollider/commit/531758a60742d3aea9f817b6c2c76621c5ce0c17 so I would be surprised if this has been not working properly for 20 years?

Reply:
I don't know, I was just following my nose. I will dig into it. Perhaps the version of Bonjour is relevant, but I tried on Intel and M1 and the published port is incorrectly byte-swapped for both.

Reply:
I think there's evidence that this worked at some point for at least someone in the past

https://github.com/supercollider/supercollider/blob/917faf1a8d2d6dc7c7a753590307518085a7419a/HelpSource/Tutorials/Mark_Polishook_tutorial/01_Startup.schelp#L38C27-L38C32